src/ModelFactory.php

```diff
@@ -18,13 +15,13 @@ public function __construct()
         parent::__construct(static::getClass());
     }
 
-    public static function __callStatic(string $name, array $arguments)
```

Comment:
Im not sure why this method was "magic" from the beginning.

Reply:
This is because there used to be a `createMany` instance method on `Factory` that we deprecated (ref #111).

Reply:
I looked into this a bit further to see why in #111 we didn't do this. There was a typo in a test proving this `__callStatic()` is required (see #187).

Once I merge #187 and you rebase this PR, this test will fail. So we do need to keep `__callStatic()` (until 2.0 anyway).

Reply:
There already is a non-static `createMany` method, so we cannot define it normally (like you tried to do now). The BC layer has to use the magic method.

Reply:
There must be a `createMany` static and a `createMany` non-static method (in `Factory`). The non-static one is deprecated. Because of this, we have to use `__callStatic` and `__call` magic and cannot define the method directly afaics.

Reply:
I spent over an hour on this. 

There is no way for PHP to detect if you call a static method in an non-static way. The current implementation (with magic methods) is the only way to detect that.